source/extensions/filters/network/thrift_proxy/buffer_helper.cc

```diff
@@ -0,0 +1,200 @@
+#include "extensions/filters/network/thrift_proxy/buffer_helper.h"
+
+#include "common/common/byte_order.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+
+int8_t BufferHelper::peekI8(Buffer::Instance& buffer, size_t offset) {
+  ASSERT(buffer.length() >= 1);
+  int8_t i;
+  buffer.copyOut(offset, 1, &i);
```

Comment:
It might be worthwhile to add special methods to `Buffer::Instance` for the 1, 2, and 4 byte special cases, since a general-purpose copy will be slow for those. But I’m happy to wait and see if this shows up as an issue in profiles first.

Reply:
I think I'll hold off for now. If it's useful, we'll want to use across a bunch of protocols and it should go in its own PR.